### PR TITLE
fix(gatsby): fix error cannot read referencePath of undefined

### LINKED ISSUE #20488: api-runner-node.js: TypeError: Cannot read property 'referencePaths' of undefined
## Description

Describe the issue that you're seeing.

### Steps to reproduce

- checkout gatsby
- `cd www`
- install packages
- run `gatsby develop`

### Expected result

no error

### Actual result

error:

```shell
parsing error {
  file: 'gatsby/src/utils/api-runner-node.js',
  e: TypeError: Cannot read property 'referencePaths' of undefined
      at CallExpression (/Users/muescha/Work/gatsby/github/gatsby/www/plugins/gatsby-transformer-gatsby-api-calls/find-api-calls.js:123:52)
      at NodePath._call (/Users/muescha/Work/gatsby/github/gatsby/www/plugins/gatsby-transformer-gatsby-api-calls/node_modules/@babel/traverse/lib/path/context.js:55:20)
      at NodePath.call (/Users/muescha/Work/gatsby/github/gatsby/www/plugins/gatsby-transformer-gatsby-api-calls/node_modules/@babel/traverse/lib/path/context.js:42:17)
      at NodePath.visit (/Users/muescha/Work/gatsby/github/gatsby/www/plugins/gatsby-transformer-gatsby-api-calls/node_modules/@babel/traverse/lib/path/context.js:90:31)
      at TraversalContext.visitQueue (/Users/muescha/Work/gatsby/github/gatsby/www/plugins/gatsby-transformer-gatsby-api-calls/node_modules/@babel/traverse/lib/context.js:112:16)
      at TraversalContext.visitSingle (/Users/muescha/Work/gatsby/github/gatsby/www/plugins/gatsby-transformer-gatsby-api-calls/node_modules/@babel/traverse/lib/context.js:84:19)
      at TraversalContext.visit (/Users/muescha/Work/gatsby/github/gatsby/www/plugins/gatsby-transformer-gatsby-api-calls/node_modules/@babel/traverse/lib/context.js:140:19)
      at Function.traverse.node (/Users/muescha/Work/gatsby/github/gatsby/www/plugins/gatsby-transformer-gatsby-api-calls/node_modules/@babel/traverse/lib/index.js:84:17)
      at NodePath.visit (/Users/muescha/Work/gatsby/github/gatsby/www/plugins/gatsby-transformer-gatsby-api-calls/node_modules/@babel/traverse/lib/path/context.js:97:18)
      at TraversalContext.visitQueue (/Users/muescha/Work/gatsby/github/gatsby/www/plugins/gatsby-transformer-gatsby-api-calls/node_modules/@babel/traverse/lib/context.js:112:16)
      at TraversalContext.visitMultiple (/Users/muescha/Work/gatsby/github/gatsby/www/plugins/gatsby-transformer-gatsby-api-calls/node_modules/@babel/traverse/lib/context.js:79:17)
      at TraversalContext.visit (/Users/muescha/Work/gatsby/github/gatsby/www/plugins/gatsby-transformer-gatsby-api-calls/node_modules/@babel/traverse/lib/context.js:138:19)
      at Function.traverse.node (/Users/muescha/Work/gatsby/github/gatsby/www/plugins/gatsby-transformer-gatsby-api-calls/node_modules/@babel/traverse/lib/index.js:84:17)
      at NodePath.visit (/Users/muescha/Work/gatsby/github/gatsby/www/plugins/gatsby-transformer-gatsby-api-calls/node_modules/@babel/traverse/lib/path/context.js:97:18)
      at TraversalContext.visitQueue (/Users/muescha/Work/gatsby/github/gatsby/www/plugins/gatsby-transformer-gatsby-api-calls/node_modules/@babel/traverse/lib/context.js:112:16)
      at TraversalContext.visitMultiple (/Users/muescha/Work/gatsby/github/gatsby/www/plugins/gatsby-transformer-gatsby-api-calls/node_modules/@babel/traverse/lib/context.js:79:17)
      at TraversalContext.visit (/Users/muescha/Work/gatsby/github/gatsby/www/plugins/gatsby-transformer-gatsby-api-calls/node_modules/@babel/traverse/lib/context.js:138:19)
      at Function.traverse.node (/Users/muescha/Work/gatsby/github/gatsby/www/plugins/gatsby-transformer-gatsby-api-calls/node_modules/@babel/traverse/lib/index.js:84:17)
      at NodePath.visit (/Users/muescha/Work/gatsby/github/gatsby/www/plugins/gatsby-transformer-gatsby-api-calls/node_modules/@babel/traverse/lib/path/context.js:97:18)
      at TraversalContext.visitQueue (/Users/muescha/Work/gatsby/github/gatsby/www/plugins/gatsby-transformer-gatsby-api-calls/node_modules/@babel/traverse/lib/context.js:112:16)
      at TraversalContext.visitSingle (/Users/muescha/Work/gatsby/github/gatsby/www/plugins/gatsby-transformer-gatsby-api-calls/node_modules/@babel/traverse/lib/context.js:84:19)
      at TraversalContext.visit (/Users/muescha/Work/gatsby/github/gatsby/www/plugins/gatsby-transformer-gatsby-api-calls/node_modules/@babel/traverse/lib/context.js:140:19)
}
```

### Environment

Run `gatsby info --clipboard` in your project directory and paste the output here:

```
  System:
    OS: macOS Mojave 10.14.5
    CPU: (8) x64 Intel(R) Core(TM) i7-3820QM CPU @ 2.70GHz
    Shell: 3.2.57 - /bin/bash
  Binaries:
    Node: 12.13.1 - ~/.nvm/versions/node/v12.13.1/bin/node
    Yarn: 1.21.1 - /usr/local/bin/yarn
    npm: 6.13.1 - ~/.nvm/versions/node/v12.13.1/bin/npm
  Languages:
    Python: 2.7.10 - /usr/bin/python
  Browsers:
    Chrome: 79.0.3945.88
    Safari: 12.1.1
  npmPackages:
    gatsby: ^2.18.8 => 2.18.18 
    gatsby-design-tokens: ~1.0.10 => 1.0.10 
    gatsby-image: ^2.2.34 => 2.2.37 
    gatsby-plugin-canonical-urls: ^2.1.16 => 2.1.18 
    gatsby-plugin-catch-links: ^2.1.19 => 2.1.21 
    gatsby-plugin-emotion: ^4.1.16 => 4.1.18 
    gatsby-plugin-feed: ^2.3.23 => 2.3.25 
    gatsby-plugin-google-analytics: ^2.1.29 => 2.1.31 
    gatsby-plugin-google-tagmanager: ^2.1.18 => 2.1.20 
    gatsby-plugin-guess-js: ^1.1.26 => 1.1.28 
    gatsby-plugin-layout: ^1.1.16 => 1.1.18 
    gatsby-plugin-mailchimp: ^2.2.3 => 2.2.3 
    gatsby-plugin-manifest: ^2.2.31 => 2.2.34 
    gatsby-plugin-mdx: ^1.0.59 => 1.0.64 
    gatsby-plugin-netlify: ^2.1.28 => 2.1.30 
    gatsby-plugin-netlify-cache: ^0.1.0 => 0.1.0 
    gatsby-plugin-nprogress: ^2.1.15 => 2.1.17 
    gatsby-plugin-offline: ^3.0.27 => 3.0.30 
    gatsby-plugin-react-helmet: ^3.1.16 => 3.1.18 
    gatsby-plugin-sharp: ^2.3.5 => 2.3.10 
    gatsby-plugin-sitemap: ^2.2.22 => 2.2.24 
    gatsby-plugin-theme-ui: ^0.2.43 => 0.2.43 
    gatsby-plugin-twitter: ^2.1.15 => 2.1.17 
    gatsby-plugin-typography: ^2.3.18 => 2.3.20 
    gatsby-remark-autolink-headers: ^2.1.19 => 2.1.22 
    gatsby-remark-code-titles: ^1.1.0 => 1.1.0 
    gatsby-remark-copy-linked-files: ^2.1.31 => 2.1.33 
    gatsby-remark-embedder: ^1.6.0 => 1.7.0 
    gatsby-remark-graphviz: ^1.1.18 => 1.1.20 
    gatsby-remark-images: ^3.1.35 => 3.1.39 
    gatsby-remark-normalize-paths: ^1.0.0 => 1.0.0 
    gatsby-remark-prismjs: ^3.3.25 => 3.3.29 
    gatsby-remark-responsive-iframe: ^2.2.28 => 2.2.30 
    gatsby-remark-smartypants: ^2.1.17 => 2.1.19 
    gatsby-source-airtable: ^2.0.12 => 2.0.12 
    gatsby-source-filesystem: ^2.1.40 => 2.1.43 
    gatsby-source-npm-package-search: ^2.1.19 => 2.1.21 
    gatsby-transformer-csv: ^2.1.19 => 2.1.21 
    gatsby-transformer-documentationjs: ^4.1.20 => 4.1.22 
    gatsby-transformer-remark: ^2.6.39 => 2.6.45 
    gatsby-transformer-screenshot: ^2.1.44 => 2.1.46 
    gatsby-transformer-sharp: ^2.3.7 => 2.3.9 
    gatsby-transformer-yaml: ^2.2.18 => 2.2.20 
  npmGlobalPackages:
    gatsby-cli: 2.8.13
```

### PR DESCRIPTION
**Pull request by muescha**
## Description

since the file `api-runner-node-error-parser.js` is an helper function and not need a link to the source i assume it can be fixed with a "monkey patch": renaming the file:

from: `api-runner-node-error-parser.js`
to: `api-runner-error-parser.js`

see #20488

## Related Issues

fixes #20488

### CODE REVIEW

File: packages/gatsby/src/utils/api-runner-node.js [hunk 1]
```diff
@@ -22,7 +22,9 @@ const { emitter, store } = require(`../redux`)
 const getPublicPath = require(`./get-public-path`)
 const { getNonGatsbyCodeFrameFormatted } = require(`./stack-trace-utils`)
 const { trackBuildError, decorateEvent } = require(`gatsby-telemetry`)
-const { default: errorParser } = require(`./api-runner-node-error-parser`)
+const {
```
Comment (sidharthachatterjee) [inline packages/gatsby/src/utils/api-runner-node.js]: const { default: errorParser } = require(`./api-runner-error-parser`)

Comment (muescha) [reply packages/gatsby/src/utils/api-runner-node.js]: that is my prettier :/
try to fix it

Comment (muescha) [reply packages/gatsby/src/utils/api-runner-node.js]: fixed

**Comment by adrienharnay** [comment]
I (kind of) have the same error:

```
TypeError: Cannot read property 'match' of undefined
  
  - api-runner-node-error-parser.js:39 errorParser
    [front-end]/[gatsby]/dist/utils/api-runner-node-error-parser.js:39:33
  
  - api-runner-node.js:398 
    [front-end]/[gatsby]/dist/utils/api-runner-node.js:398:29
```

But applying the changes from this PR doesn't seem to fix it... Would you an idea?

Thanks in advance

**Comment by muescha** [comment]
Thats a different error

This PR fix an error in `find-api-calls.js` (See #20488)

**Comment by muescha** [comment]
It would be very helpful if you open an new issue with the new error

**Comment by shicholas** [comment]
> I (kind of) have the same error:
> 
> ```
> TypeError: Cannot read property 'match' of undefined
>   
>   - api-runner-node-error-parser.js:39 errorParser
>     [front-end]/[gatsby]/dist/utils/api-runner-node-error-parser.js:39:33
>   
>   - api-runner-node.js:398 
>     [front-end]/[gatsby]/dist/utils/api-runner-node.js:398:29
> ```
> 
> But applying the changes from this PR doesn't seem to fix it... Would you an idea?
> 
> Thanks in advance

@adrienharnay, I had something similar then managed to get it working with a fresh package install and deleting the `.cache` directory. I hope that helps.

**Comment by muescha** [comment]
@adrienharnay  please look at PR #20730 - maybe this PR fix your problem

**Comment by adrienharnay** [comment]
Hey, actually it was an opaque error from gatsby-source-graphql... So not related, sorry.

**Pull request merged**